common/ManagementTestShared/ManagementRecordedTestBase.cs

```diff
@@ -25,18 +25,6 @@ protected ManagementRecordedTestBase(bool isAsync, RecordedTestMode mode) : base
         {
```

Comment:
What is the expected difference between ManagementRecordedTestBase and your newly added ManagementRecordTestBase? Should they be the same class? If not when should someone use one over the other?

Reply:
The ideal architecture is for all of this is moved to /sdk/core/Azure.Core.TestFramework/src/ as @pakrym has suggested in other meetings. This would make the PR larger, though.

I would then want to name the new class something like: /sdk/core/Azure.Core.TestFramework/src/ OneTimeSetupRecordBase.cs

This would mean we move : /common/ManagementTestShared/ResourceGroupCleanupPolicy.cs

over to sdk/core as well.

We could then use ManagementRecordedTestBase.cs (which is now in sdk/core/src) in the following way:

assuming we do not consolidate OneTimeSetupRecordBase and RecordedTestBase.cs, then ManagementRecordedTestBase.cs cannot inherit from RecordedTestBase anymore. It will have to move down into a common class OneTimeSetupRecordBase and RecordedTestBase share (this may make the most sense), or have to have some property that holds an instance of OneTimeSetupRecordBase or RecordedTestbase which is not wanted I think.

If we merge RecordedTestBase and OneTimeSetupRecordBase , then this won't really be an issue. I know @pakrym is for merging these two, but I think the separation make it clear to a user what they are doing. 

Can also make ManagementRecordedTestBase not inherited from anything and instead by a common helper class shared between OneTimeSetupRecordBase  and RecordedTestBase with the need variables passed into method calls as needed. Also, doesn't seem like the best approach though.